crates/cast/bin/cmd/storage.rs

```diff
@@ -202,23 +202,52 @@ async fn fetch_storage_slots(
     address: NameOrAddress,
     layout: &StorageLayout,
 ) -> Result<Vec<B256>> {
-    // TODO: Batch request
-    let futures: Vec<_> = layout
-        .storage
-        .iter()
-        .map(|slot| {
-            let slot = B256::from(U256::from_str(&slot.slot)?);
-            Ok(provider.get_storage_at(address.clone(), slot.to_ethers(), None))
-        })
-        .collect::<Result<_>>()?;
-
-    join_all(futures).await.into_iter().map(|r| Ok(r?.to_alloy())).collect()
+    let mut results = Vec::new();
+
+    for storage_entry in &layout.storage {
```

Comment:
`storage_entry` already has the offset included, right?
so we need to use this to offset into the slot